dvc/objects/cache.py

```diff
@@ -0,0 +1,54 @@
+import pickle
+from functools import wraps
+from typing import Any
+
+import diskcache
+from diskcache import Disk as disk
+from diskcache import Index  # noqa: F401, pylint: disable=unused-import
+
+
+class DiskError(Exception):
+    def __init__(self, directory: str) -> None:
+        self.directory = directory
+        super().__init__(f"Could not open disk: {directory}")
+
+
+def translate_pickle_error(fn):
+    @wraps(fn)
+    def wrapped(self, *args, **kwargs):
+        try:
+            return fn(self, *args, **kwargs)
+        except (pickle.PickleError, ValueError) as exc:
+            if isinstance(exc, ValueError) and not str(exc).startswith(
```

Comment:
based on https://github.com/iterative/dvc/issues/7726 it looks like we need to just do a substring search for `pickle`, apparently there are `ValueError`s raised in 3.7 w/messages that don't start with `pickle protocol`

Reply:
Thanks @pmrowla. Done in a66425f. We now search for  `pickle protocol` substring.